Scripts/Spells: Stealth, Subterfuge and Subterfuge effect

Daniel25 opened the pull request:
-  Stealh working, it wasnt working.
-  Subterfuge working, it wasnt working.

- [x] master
- last commit: https://github.com/TrinityCore/TrinityCore/commit/52c7f12ed35f321dc97d8caea2b112287f5ac05b

**It builds and works in game.

Daniel25 committed:
Stealth - 1784, Subterfuge - 115191 and Subterfuge effect - 115192

Daniel25 committed:
Update spell_rogue.cpp

joschiwald commented on src/server/scripts/Spells/spell_rogue.cpp:
```diff
@@ -602,65 +607,85 @@ class spell_rog_shiv : public SpellScriptLoader
-        {
-            return new spell_rog_stealth_AuraScript();
-        }
+    AuraScript* GetAuraScript() const
```
bad indentation, bad override removal

joschiwald commented on src/server/scripts/Spells/spell_rogue.cpp:
```diff
@@ -602,65 +607,85 @@ class spell_rog_shiv : public SpellScriptLoader
         {
             PrepareAuraScript(spell_rog_stealth_AuraScript);
 
-            bool Validate(SpellInfo const* /*spellInfo*/) override
```
it is bad to remove this validate, you need to update the spells

joschiwald commented on src/server/scripts/Spells/spell_rogue.cpp:
```diff
@@ -602,65 +607,85 @@ class spell_rog_shiv : public SpellScriptLoader
 };
 
-// 1784 - Stealth
+/// Stealth - 1784, Subterfuge - 115191 and Subterfuge effect - 115192
```
don't change this comment if you not bind the spellscript to all these 3 spells

joschiwald commented on src/server/scripts/Spells/spell_rogue.cpp:
```diff
@@ -32,6 +32,7 @@
 #include "SpellHistory.h"
 #include "SpellMgr.h"
 #include "SpellScript.h"
+#include "GridNotifiers.h"
```
why?

joschiwald commented on src/server/scripts/Spells/spell_rogue.cpp:
```diff
@@ -602,65 +607,85 @@ class spell_rog_shiv : public SpellScriptLoader
+                    }
+                    else
+                    {
+                        if (l_Caster->HasAura(eSpells::StealthSubterfuge))
```
useless hasaura check

HannibalRoG commented:
This PR close https://github.com/TrinityCore/TrinityCore/issues/19712

Daniel25 committed:
Update spell_rogue.cpp

Daniel25 committed:
Update spell_rogue.cpp

Daniel25 committed:
Update spell_rogue.cpp

Daniel25 committed:
Update spell_rogue.cpp

joschiwald commented on src/server/scripts/Spells/spell_rogue.cpp:
```diff
@@ -640,15 +652,11 @@ class spell_rog_stealth : public SpellScriptLoader
-                    else
-                    {
-                        if (l_Caster->HasAura(eSpells::StealthSubterfuge))
-                            l_Caster->RemoveAura(eSpells::StealthSubterfuge);
```
why do you remove the `RemoveAura` now?

Daniel25 replied:
Sorry Im doing too much things in the same time...

Daniel25 committed:
Update spell_rogue.cpp

joschiwald commented on src/server/scripts/Spells/spell_rogue.cpp:
```diff
@@ -622,38 +635,61 @@ class spell_rog_stealth : public SpellScriptLoader
+                        l_Caster->CastSpell(l_Caster, eSpells::StealthTriggered1, true);
+                        l_Caster->CastSpell(l_Caster, eSpells::StealthTriggered2, true);
 
-                // Master of Subtlety
```
what happend with Master of Subtlety?

Daniel25 committed:
Update spell_rogue.cpp

Daniel25 committed:
Update spell_rogue.cpp

Daniel25 committed:
Update spell_rogue.cpp

Daniel25 committed:
Update spell_rogue.cpp

joschiwald commented on src/server/scripts/Spells/spell_rogue.cpp:
```diff
@@ -621,33 +634,72 @@ class spell_rog_stealth : public SpellScriptLoader
             void HandleEffectApply(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
             {
-                Unit* target = GetTarget();
+                if (Unit* target = GetTarget())
```
target never null

Aokromes commented on src/server/scripts/Spells/spell_rogue.cpp:
```diff
@@ -657,10 +709,10 @@ class spell_rog_stealth : public SpellScriptLoader
-        {
-            return new spell_rog_stealth_AuraScript();
-        }
+		AuraScript* GetAuraScript() const
```
tabs2spaces.

Daniel25 committed:
Update spell_rogue.cpp

Daniel25 committed:
Update spell_rogue.cpp

Aokromes commented on src/server/scripts/Spells/spell_rogue.cpp:
```diff
@@ -642,11 +642,11 @@ class spell_rog_stealth : public SpellScriptLoader
             }
 
-            void HandleEffectApply(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+	    void HandleEffectApply(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
```
tab2spaces.

Aokromes commented on src/server/scripts/Spells/spell_rogue.cpp:
```diff
@@ -607,6 +611,15 @@ class spell_rog_stealth : public SpellScriptLoader
 {
     public:
         spell_rog_stealth() : SpellScriptLoader("spell_rog_stealth") { }
+        
```
remove whitespaces.

Aokromes commented on src/server/scripts/Spells/spell_rogue.cpp:
```diff
@@ -621,33 +634,72 @@ class spell_rog_stealth : public SpellScriptLoader
             void HandleEffectRemove(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
             {
-                Unit* target = GetTarget();
+	    	Unit* target = GetTarget();
```
tab2spaces.

Aokromes commented on src/server/scripts/Spells/spell_rogue.cpp:
```diff
@@ -621,33 +634,72 @@ class spell_rog_stealth : public SpellScriptLoader
 
-                target->RemoveAurasDueToSpell(SPELL_ROGUE_STEALTH_STEALTH_AURA);
-                target->RemoveAurasDueToSpell(SPELL_ROGUE_STEALTH_SHAPESHIFT_AURA);
+		   target->RemoveAurasDueToSpell(SPELL_ROGUE_STEALTH_STEALTH_AURA);
```
tab2spaces

Aokromes commented on src/server/scripts/Spells/spell_rogue.cpp:
```diff
@@ -657,10 +709,10 @@ class spell_rog_stealth : public SpellScriptLoader
-        {
-            return new spell_rog_stealth_AuraScript();
-        }
+	AuraScript* GetAuraScript() const
```
tab2spaces

Daniel25 replied:
Sorry dunno what is going on with whitespaces. Its ok in my screen but its broken then... Does someone care in editing that mistakes please?

Daniel25 committed:
Update spell_rogue.cpp

Aokromes commented on src/server/scripts/Spells/spell_rogue.cpp:
```diff
@@ -605,9 +609,18 @@ class spell_rog_shiv : public SpellScriptLoader
 class spell_rog_stealth : public SpellScriptLoader
 {
-    public:
+     public:
```
remove aditional whitespaces, we use 4 spaces, not 5.

Aokromes commented on src/server/scripts/Spells/spell_rogue.cpp:
```diff
@@ -605,9 +609,18 @@ class spell_rog_shiv : public SpellScriptLoader
+     public:
         spell_rog_stealth() : SpellScriptLoader("spell_rog_stealth") { }
 
+       	enum eSpells
```
tab2spaces

Aokromes commented on src/server/scripts/Spells/spell_rogue.cpp:
```diff
@@ -621,33 +634,72 @@ class spell_rog_stealth : public SpellScriptLoader
-                target->CastSpell(target, SPELL_ROGUE_STEALTH_STEALTH_AURA, TRIGGERED_FULL_MASK);
-                target->CastSpell(target, SPELL_ROGUE_STEALTH_SHAPESHIFT_AURA, TRIGGERED_FULL_MASK);
-            }
+	    	Unit* target = GetTarget();
```
tabs2spaces

Aokromes commented on src/server/scripts/Spells/spell_rogue.cpp:
```diff
@@ -621,33 +634,72 @@ class spell_rog_stealth : public SpellScriptLoader
             void HandleEffectRemove(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
             {
-                Unit* target = GetTarget();
+		Unit* target = GetTarget();
```
tab2spaces

Daniel25 committed:
Update spell_rogue.cpp

Daniel25 committed:
Update spell_rogue.cpp

joschiwald commented:
stop doing random commits, it looks like you don't know what you are doing

just replace your tabs with 4 spaces

Daniel25 committed:
Update spell_rogue.cpp

Daniel25 commented:
Well i said it before....dunno what is going on with my screen but its ok and when i edited it, it isnt....

joschiwald commented:
exactly, you insert spaces and tabs randomly to looks fine on your screen, you need to know what you are inserting, don't use the github editor, use some editor like notepad++ and enable display of tabs and spaces

Daniel25 commented:
I was using notepadd++ and when I copy it in the github editor its giving the troubles....sorry

Daniel25 committed:
Update spell_rogue.cpp

Daniel25 commented:
Do you care in editing it please? We will finish before if someone can do it in my place....

Aokromes commented on src/server/scripts/Spells/spell_rogue.cpp:
```diff
@@ -608,6 +612,15 @@ class spell_rog_stealth : public SpellScriptLoader
     public:
         spell_rog_stealth() : SpellScriptLoader("spell_rog_stealth") { }
 
+       	enum eSpells
```
tabs2spaces

Aokromes commented on src/server/scripts/Spells/spell_rogue.cpp:
```diff
@@ -621,34 +634,73 @@ class spell_rog_stealth : public SpellScriptLoader
             }
 
-            void HandleEffectApply(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+	  void HandleEffectApply(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
```
tabs2spaces

Aokromes commented on src/server/scripts/Spells/spell_rogue.cpp:
```diff
@@ -621,34 +634,73 @@ class spell_rog_stealth : public SpellScriptLoader
+	    void HandleEffectRemove(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
             {
-                Unit* target = GetTarget();
+		Unit* target = GetTarget();
```
tabs2spaces

Aokromes commented on src/server/scripts/Spells/spell_rogue.cpp:
```diff
@@ -621,34 +634,73 @@ class spell_rog_stealth : public SpellScriptLoader
-                target->RemoveAurasDueToSpell(SPELL_ROGUE_STEALTH_STEALTH_AURA);
-                target->RemoveAurasDueToSpell(SPELL_ROGUE_STEALTH_SHAPESHIFT_AURA);
-            }
+		target->RemoveAurasDueToSpell(SPELL_ROGUE_STEALTH_STEALTH_AURA);
```
tabs2spaces

Daniel25 committed:
Update spell_rogue.cpp

Daniel25 commented:
It isnt working now...dont merge it still....

Daniel25 commented:
With all that changes the fix doesnt work now....So dont merge it still...I have to check it again....

Daniel25 committed:
Update spell_rogue.cpp

Validate funtion is commented, it has any issue and the fix doesnt work if it isnt commented.

Daniel25 commented:
Validate funtion is commented because the fix doesnt work with the funtion active. I guess that its what broken the stealth spell before.
It compiles and works good now.

joschiwald commented on src/server/scripts/Spells/spell_rogue.cpp:
```diff
@@ -605,62 +611,105 @@ class spell_rog_shiv : public SpellScriptLoader
-    public:
-        spell_rog_stealth() : SpellScriptLoader("spell_rog_stealth") { }
+public:
+    spell_rog_stealth() : SpellScriptLoader("spell_rog_stealth") { }
```
pls restore the original indentation

Daniel25 committed:
Update spell_rogue.cpp

Daniel25 commented:
Thanks you for the patiente and sorry for the mistakes. Its my first pull request :/.

joschiwald commented on src/server/scripts/Spells/spell_rogue.cpp:
```diff
@@ -612,23 +618,46 @@ class spell_rog_stealth : public SpellScriptLoader
             void HandleEffectApply(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
             {
                 Unit* target = GetTarget();
+                if (Unit* target = GetTarget())
```
useless if condition

joschiwald commented on src/server/scripts/Spells/spell_rogue.cpp:
```diff
@@ -641,7 +670,27 @@ class spell_rog_stealth : public SpellScriptLoader
             void HandleEffectRemove(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
             {
                 Unit* target = GetTarget();
+                if (Unit* target = GetTarget())
```
useless if condition

joschiwald commented on src/server/scripts/Spells/spell_rogue.cpp:
```diff
@@ -657,7 +706,7 @@ class spell_rog_stealth : public SpellScriptLoader
         };
 
-        AuraScript* GetAuraScript() const override
+        AuraScript* GetAuraScript() const
```
you removed override keyword again

joschiwald commented on src/server/scripts/Spells/spell_rogue.cpp:
```diff
@@ -641,7 +670,27 @@ class spell_rog_stealth : public SpellScriptLoader
+                    {
+                        target->RemoveAura(SPELL_ROGUE_STEALTH_SHAPESHIFT_AURA);
+
+                        if (Aura* l_Nightstalker = target->GetAura(SPELL_ROGUE_NIGHTSTALKER_DAMAGE_DONE))
```
`nightstalker`

joschiwald commented on src/server/scripts/Spells/spell_rogue.cpp:
```diff
@@ -641,7 +670,27 @@ class spell_rog_stealth : public SpellScriptLoader
+                        target->RemoveAura(SPELL_ROGUE_STEALTH_SHAPESHIFT_AURA);
+
+                        if (Aura* l_Nightstalker = target->GetAura(SPELL_ROGUE_NIGHTSTALKER_DAMAGE_DONE))
+                            l_Nightstalker->SetDuration(200);   ///< We can't remove it now
```
why?

joschiwald commented on src/server/scripts/Spells/spell_rogue.cpp:
```diff
@@ -612,23 +618,46 @@ class spell_rog_stealth : public SpellScriptLoader
                 Unit* target = GetTarget();
+                if (Unit* target = GetTarget())
+                {
+                    if (GetSpellInfo()->Id != SPELL_ROGUE_STEALTH_SUBTERFUGE_EFFECT)
```
this will never work, because the spell is not bound to this script

Daniel25 replied:
Sorry I dont know to bound it.

Daniel25 committed:
Update spell_rogue.cpp

Daniel25 committed:
Merge pull request #1 from TrinityCore/master

Daniel25 commented:
If I revert SPELL_ROGUE_STEALTH_SUBTERFUGE and SPELL_ROGUE_STEALTH_SUBTERFUGE_EFFECT to what I had in the begin,will it work?

Daniel25 committed:
Merge pull request #2 from TrinityCore/master

update_20170927

joschiwald commented:
broken code

supercedes by b12b37058c863bcfecfefc965da45f31b604ee3b
